cpp/ql/src/Best Practices/Magic Constants/MagicConstantsString.qhelp

```diff
@@ -20,7 +20,7 @@ update if the requirements change (for example, a protocol is updated).
 </li>
 </ol>
 
-<p>This rule finds magic strings for which there is no pre-existing named
+<p>This rule finds magic strings for which there is no preexisting named
```

Comment:
If indeed both are correct, then I think I also slightly prefer 'pre-existing' for the same reason, but I'll gladly defer to someone with a stronger opinion.

Reply:
> British English often prefers hyphenated compounds, such as anti-smoking, whereas American English discourages the use of hyphens in compounds where there is no compelling reason, so antismoking is much more common. [—Wikipedia](https://en.wikipedia.org/wiki/American_and_British_English_spelling_differences?wprov=sfti1)

I suppose if American English is our target dialect in our documentation [which we do, IIRC; citation needed], then it would mean prioritizing the hyphenless spelling, though I personally agree the hyphenated spelling is easier to parse.

Reply:
> citation needed

https://github.com/github/codeql/blob/main/docs/qldoc-style-guide.md#language-requirements

Reply:
Obviously we should be doing like they do in The New Yorker, and write "preëxisting"...

Jokes aside, while I agree that we should usually follow American English, I think the version with the hyphen is more readable, and I think readability trumps correctness in this case.

Reply:
Yeah, I think I'd also prefer the hyphenated version.

Reply:
I think we agree we want to keep the hyphenation, so please drop those changes.

Reply:
Dropped: https://github.com/github/codeql/compare/a931c1553d9cb079c55b8b4ff48d3487b693ece7..c92ce69f480f0b397fe6c1f2ed5ae934815b8398

Reply:
Thanks.